rust/src/lib/nm/query_apply/ip.rs

```diff
@@ -183,6 +184,17 @@ fn nm_dhcp_duid_to_nmstate(nm_setting: &NmSettingIp) -> Option<Dhcpv6Duid> {
     }
 }
 
+pub(crate) fn nm_forwarding_to_nmstate(nm: &NmSettingIp) -> Option<bool> {
+    nm.forwarding.and_then(|val| match val {
+        0 => Some(false),
+        1 => Some(true),
+        other => {
+            log::debug!("ignoring unsupported NM forwarding mode {}", other);
+            None
+        }
```

Comment:
Note that the most common value configured in NM connections will be `FORWARDING_DEFAULT (-1)`. Less common, but also likely to be configured `FORWARDING_AUTO (2)`. With this, a debug message will be shown in those cases.
What I would do is to silently convert those to None, and emit a warning if we receive unknown values, like it is done in other places of the file.